### PR TITLE
Pagination backend implementation

### PR DESCRIPTION
**Pull request by Lord-Bubbles**
#19

### CODE REVIEW

File: src/components/Header.js [hunk 1]
```diff
@@ -42,11 +42,10 @@ class Header extends React.Component {
...
 
   // Filters what gets displayed based on the user inputs
-  /*
-  fetchCurRecipes = async () => {
-    await fetch(`/recipes?page=${this.state.currentPage}&limit=${this.state.pageSize}&` +
-     `${this.createIngredientQueryParam()}name=${this.state.name}&min=${this.state.minTime}` +
-     `&max=${this.state.maxTime}&username=${this.state.createdBy}&recipes=${LZString.decompress(localStorage.getItem("recipes"))}`, {
+  fetchRecipes = () => {
+    fetch(`/recipes?page=${this.state.currentPage}&limit=${this.state.pageSize}&` +
+     `${this.createIngredientQueryParam()}name=${this.state.name}&minTime=${this.state.minTime}` +
```
Comment (chilimalone) [inline src/components/Header.js:47]: if we're keeping with 80 character limit, these lines are a bit too long. Also the indentation is weird

File: src/components/Header.js [hunk 2]
```diff
@@ -42,11 +42,10 @@ class Header extends React.Component {
   }
 
   // Filters what gets displayed based on the user inputs
-  /*
-  fetchCurRecipes = async () => {
-    await fetch(`/recipes?page=${this.state.currentPage}&limit=${this.state.pageSize}&` +
-     `${this.createIngredientQueryParam()}name=${this.state.name}&min=${this.state.minTime}` +
-     `&max=${this.state.maxTime}&username=${this.state.createdBy}&recipes=${LZString.decompress(localStorage.getItem("recipes"))}`, {
+  fetchRecipes = () => {
+    fetch(`/recipes?page=${this.state.currentPage}&limit=${this.state.pageSize}&` +
```
Comment (chilimalone) [inline src/components/Header.js:46]: usually if indentation is +2 spaces, then line breaks in code is 2x, meaning +4 spaces

File: src/components/Header.js [hunk 3]
```diff
@@ -55,19 +54,11 @@ class Header extends React.Component {
...
-        localStorage.setItem("recipes", LZString.compress(JSON.stringify(data.all)));
-      }
-    });
-    this.setState({
-      ingredients: [],
-      minTime: 0,
-      maxTime: 0,
-      createdBy: "",
-    });
+      this.setState({recipes: data.data, totalCount: data.totalCount})
```
Comment (chilimalone) [inline src/components/Header.js:57]: +4 spaces, rather than +2

File: src/controllers/RecipeController.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    query["time"] = {$gte: minTime};
+  }
+  if (ingredients) {
+    query["ingredients"] = {$elemMatch: {$in: ingredients}};
+  }
+  if (user) {
+    query["createdBy"] = user;
+  }
+  const recipes = await Recipe
+    .find(query)
```
Comment (chilimalone) [inline src/controllers/RecipeController.js:45]: +4 spaces

File: src/controllers/RecipeController.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    .exec();
+  const count = await Recipe.estimatedDocumentCount();
+  res.status(200).json({data: recipes, totalCount: count});
+});
+
+// @desc
+// @route  
+// @access 
+const addRecipe = asyncHandler(async (req, res) => {
+
```
Comment (chilimalone) [inline src/controllers/RecipeController.js:58]: add todo here

File: src/controllers/RecipeController.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    .find(query)
+    .select("_id name ingredients method imageID createdBy time")
+    .skip((page - 1) * limit)
+    .limit(limit)
+    .exec();
+  const count = await Recipe.estimatedDocumentCount();
+  res.status(200).json({data: recipes, totalCount: count});
+});
+
+// @desc
```
Comment (chilimalone) [inline src/controllers/RecipeController.js:54]: fill out?

File: src/controllers/RecipeController.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+// @route  
+// @access 
+const addRecipe = asyncHandler(async (req, res) => {
+
+});
+
+// @desc
```
Comment (chilimalone) [inline src/controllers/RecipeController.js:61]: ditto

File: src/controllers/RecipeController.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+// @route
+// @access
+const modifyRecipe = asyncHandler(async (req, res) => {
+
+});
+
+module.exports = {
+  fetchRecipes,
```
Comment (chilimalone) [inline src/controllers/RecipeController.js:76]: +4 spaces

File: src/controllers/RecipeController.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+});
+
+// @desc
+// @route
+// @access
+const deleteRecipe = asyncHandler(async (req, res) => {
+
+});
+
+// @desc
```
Comment (chilimalone) [inline src/controllers/RecipeController.js:68]: ditto fill out

**Review by chilimalone — COMMENTED** [review]
Probably also want to edit the commit message - there aren't any tests (like RecipeControllerTest.js or something. You also want the commit message to be present case. E.g. "Pagination backend implementation"

File: src/components/Header.js (hunk 3 above)
Comment (Lord-Bubbles) [reply src/components/Header.js:57]: this isn't a line break though?

Comment (Lord-Bubbles) [reply src/components/Header.js:57]: > +4 spaces, rather than +2

this isn't a line break though?

File: src/components/Header.js (hunk 1 above)
Comment (Lord-Bubbles) [reply src/components/Header.js:47]: fixed length of line and indentation

File: src/components/Header.js (hunk 2 above)
Comment (Lord-Bubbles) [reply src/components/Header.js:46]: done

File: src/controllers/RecipeController.js (hunk 4 above)
Comment (Lord-Bubbles) [reply src/controllers/RecipeController.js:45]: done

File: src/controllers/RecipeController.js (hunk 5 above)
Comment (Lord-Bubbles) [reply src/controllers/RecipeController.js:58]: done

File: src/controllers/RecipeController.js (hunk 6 above)
Comment (Lord-Bubbles) [reply src/controllers/RecipeController.js:54]: done

File: src/controllers/RecipeController.js (hunk 7 above)
Comment (Lord-Bubbles) [reply src/controllers/RecipeController.js:61]: done

File: src/controllers/RecipeController.js (hunk 8 above)
Comment (Lord-Bubbles) [reply src/controllers/RecipeController.js:76]: this isn't a line break either though

File: src/controllers/RecipeController.js (hunk 9 above)
Comment (Lord-Bubbles) [reply src/controllers/RecipeController.js:68]: done

File: src/components/Header.js [hunk 10]
```diff
@@ -42,11 +42,11 @@ class Header extends React.Component {
   }
 
   // Filters what gets displayed based on the user inputs
-  /*
-  fetchCurRecipes = async () => {
-    await fetch(`/recipes?page=${this.state.currentPage}&limit=${this.state.pageSize}&` +
-     `${this.createIngredientQueryParam()}name=${this.state.name}&min=${this.state.minTime}` +
-     `&max=${this.state.maxTime}&username=${this.state.createdBy}&recipes=${LZString.decompress(localStorage.getItem("recipes"))}`, {
+  fetchRecipes = () => {
+    fetch(`/recipes?page=${this.state.currentPage}&limit=` +
```
Comment (chilimalone) [inline src/components/Header.js:46]: +4 here instead of +2

File: src/components/Header.js [hunk 11]
```diff
@@ -42,11 +42,11 @@ class Header extends React.Component {
...
+        `${this.state.pageSize}&${this.createIngredientQueryParam()}name=` +
```
Comment (chilimalone) [inline src/components/Header.js:47]: -2, so you get +4 total (and below)

File: src/components/Header.js [hunk 12]
```diff
@@ -55,19 +55,17 @@ class Header extends React.Component {
...
-    });
-    this.setState({
-      ingredients: [],
-      minTime: 0,
-      maxTime: 0,
-      createdBy: "",
-    });
+      this.setState({recipes: data.data, totalCount: data.totalCount});
+      this.setState({
+        ingredients: [],
```
Comment (chilimalone) [inline src/components/Header.js:60]: what does this setState do? It seems to be just meaningless data?

**Review by chilimalone — APPROVED** [review]

File: src/components/Header.js (hunk 12 above)
Comment (Lord-Bubbles) [reply src/components/Header.js:60]: Originally, this was going to help with the searching part. After searching, I reset the values the user inputted, but I can reimplement/redesign this when I actually implement backend for searching and filtering recipes.

File: src/components/Header.js (hunk 10 above)
Comment (Lord-Bubbles) [reply src/components/Header.js:46]: According to Google Styles for JS, this should fall under block indentation so only +2. The below lines would be under line wrapping so +4 from first line until the next JSON block.

File: src/components/Header.js (hunk 11 above)
Comment (Lord-Bubbles) [reply src/components/Header.js:47]: ditto

**Pull request merged**